doc/moveit_task_constructor/moveit_task_constructor_tutorial.rst

```diff
@@ -61,28 +61,28 @@ shown in the right-most window. Selecting one of those solutions will start its
 Basic Concepts
 --------------
 
-The fundamental idea of MTC is that complex motion planning problems can be composed into a set of simpler subproblems.
+The fundamental idea of MTC is that complex motion planning problems can be composed into a set of simpler sub-problems.
 The top-level planning problem is specified as a **Task** while all subproblems are specified by **Stages**.
 Stages can be arranged in any arbitrary order and hierarchy only limited by the individual stages types.
 The order in which stages can be arranged is restricted by the direction in which results are passed.
 There are three possible stages relating to the result flow: generator, propagator, and connector stages:
 
-**Generators** compute their results independent of their neighbor stages and pass it in both directions, backwards and forwards.
+**Generators** compute their results independently of their neighbor stages and pass them in both directions, backwards and forwards.
 An example is an IK sampler for geometric poses where approaching and departing motions (neighbor stages) depend on the solution.
 
-**Propagators** receive the result of one neighbor stage, solve a subproblem and then propagate their result on to the neighbor on the opposite site.
-Depending on the implementation propagating stages can pass solutions forward, backward or in both directions separately.
+**Propagators** receive the result of one neighbor stage, solve a subproblem and then propagate their result to the neighbor on the opposite site.
+Depending on the implementation, propagating stages can pass solutions forward, backward or in both directions separately.
 An example is a stage that computes a Cartesian path based on either a start or a goal state.
 
-**Connectors** are don’t propagate any results, but rather attempt to bridge the gap between the resulting states of both neighbors.
+**Connectors** don’t propagate any results, but rather attempt to bridge the gap between the resulting states of both neighbors.
 An example is the computation of a free-motion plan from one given state to another.
 
 Additional to the order types, there are different hierarchy types allowing to encapsulate subordinate stages.
 Stages without subordinate stages are called **primitive stages**, higher-level stages are called **container stages**.
 There are three container types:
 
 **Wrappers** encapsulate a single subordinate stage and modify or filter the results.
-An example is a filter stage that only accepts solutions of an IK solver that satisfy a certain constraint.
+An example is a filter stage that only accepts solutions of an IK solver that satisfy a certain constraint. Or when you gets a generated grasp pose and you want only one that match the following IK.
```

Comment:
This extension is not clear to me.

Reply:
Wrappers are also used when you use grasp-pose generator. You pass the generated pose to the ComputeIK wrapper so that only the poses that matches the computed IK are retained. 
I found the existing example unclear myself but I understand mine can be confusing. 
We can delete it if you found it unecessary

Reply:
@YannickRiou please have a look at the alternative sentence we added in https://github.com/ros-planning/moveit_tutorials/pull/473 and add comments there if you think additional explanations are useful.